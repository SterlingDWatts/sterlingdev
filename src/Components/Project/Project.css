.Project {
  width: 100%;
  max-width: 912px;
  margin-left: auto;
  margin-right: auto;
  box-shadow: 0 1px 10px 3px rgba(0, 0, 0, 0.15),
    0 1px 4px 1px rgba(0, 0, 0, 0.15);
}

.Project img {
  width: 100%;
}

.Project--info {
  background-color: #ffff6b;
  padding: 16px;
}

.Project h3 {
  font-size: 24px;
  font-weight: 600;
  line-height: 1;
  color: #00685b;
}

.Project a {
  color: #00685b;
  font-weight: 600;
}

.Project--links {
  margin-bottom: 16px;
}

.Project--tech-icon {
  font-size: 32px;
  color: #00685b;
}

.Project__icons {
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
  margin-top: 16px;
}

.Project + .Project {
  margin-top: 16px;
}

@media screen and (min-width: 500px) {
  .Project {
    width: 80%;
    margin-left: auto;
    margin-right: auto;
  }

  .Project + .Project {
    margin-top: 24px;
  }
}
