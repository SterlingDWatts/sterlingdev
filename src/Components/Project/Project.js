import React, { Component } from "react";
import "./Project.css";

class Project extends Component {
  render() {
    return (
      <div className="Project">
        <img src={this.props.picture} alt={this.props.alt} />
        <div className="Project--info">
          <h3>{this.props.name}</h3>
          <div className="Project--links">
            <a href={this.props.siteLink}>Site</a> ∙{" "}
            <a href={this.props.clientLink}>Client Repo</a>
            {this.props.serverRepoLink && (
              <>
                {" ∙ "}
                <a href={this.props.serverRepoLink}>Server Repo</a>
              </>
            )}
          </div>
          {this.props.children}
        </div>
      </div>
    );
  }
}

export default Project;
