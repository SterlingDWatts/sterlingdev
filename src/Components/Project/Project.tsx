import React from "react";
import "./Project.css";

type AppProps = {
  picture: string;
  alt: string;
  name: string;
  siteLink: string;
  clientLink: string;
  children: any;
  serverRepoLink?: string;
};

function Project({
  picture,
  alt,
  name,
  siteLink,
  clientLink,
  children,
  serverRepoLink,
}: AppProps) {
  return (
    <div className="Project">
      <img src={picture} alt={alt} />
      <div className="Project--info">
        <h3>{name}</h3>
        <div className="Project--links">
          <a href={siteLink} rel="noopener noreferrer" target="_blank">
            Site
          </a>{" "}
          ∙{" "}
          <a href={clientLink} rel="noopener noreferrer" target="_blank">
            Client Repo
          </a>
          {serverRepoLink && (
            <>
              {" ∙ "}
              <a
                href={serverRepoLink}
                rel="noopener noreferrer"
                target="_blank"
              >
                Server Repo
              </a>
            </>
          )}
        </div>
        {children}
      </div>
    </div>
  );
}

export default Project;
