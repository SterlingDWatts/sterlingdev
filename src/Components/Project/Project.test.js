import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import Project from "./Project";

describe("Project Component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <Project
          picture="https://www.pic.com/pic.png"
          alt="project picture"
          name="project"
          siteLink="https://www.site.com"
          clientLink="https://www.site.com"
        />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the UI as expected", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Project
            picture="https://www.pic.com/pic.png"
            alt="project picture"
            name="project"
            siteLink="https://www.site.com"
            clientLink="https://www.site.com"
          />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
