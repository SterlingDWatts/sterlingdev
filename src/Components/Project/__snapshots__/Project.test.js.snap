// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project Component renders the UI as expected 1`] = `
<div
  className="Project"
>
  <img />
  <div
    className="Project--info"
  >
    <h3 />
    <div
      className="Project--links"
    >
      <a
        rel="noopener noreferrer"
        target="_blank"
      >
        Site
      </a>
       
      âˆ™
       
      <a
        rel="noopener noreferrer"
        target="_blank"
      >
        Client Repo
      </a>
    </div>
  </div>
</div>
`;
